### ISSUE TITLE
Filter tests & run with non-default provision option could be better documented

### ISSUE THREAD

**Issue by befeleme** [psss/tmt#737]
When you want to run all tests in [container](https://tmt.readthedocs.io/en/latest/examples.html#provision-options), you run 
```
tmt run -a provision --how=container <...>
```

When you want to run only [selected plan](https://tmt.readthedocs.io/en/latest/examples.html#select-plans) named *smoke*, you run:
```
tmt run plan --name smoke
``` 
which runs the tests in virtual machine.

I found no example in the documentation how to combine the two to achieve running filtered tests in container. 
It's very easy to not notice the mandatory `-a` and try running very similar command to the previous one:
```
tmt run plan --name smoke provision --how=container <...>
```
This, however, provides a container but doesn't run any tests, so there is no clear indication what was done incorrectly. NB: the container stays up and running until you manually kill it, so you don't want to make this mistake too often.

The correct command is tricky, as it requires not only adding the provision part at the end, but also the `-a` option in the middle, compare:
```
tmt run plan --name smoke
tmt run -a plan --name smoke provision --how=container <...>
```

----
Adding the more complex examples to documentation would be of a great help.
It also feels uncomfortable to have to tweak with the command in two places to achieve one change and probably more people will fall in this trap. Maybe it would be possible to run it with `-a` after the filtered part? Like this: 
```
tmt run plan --name smoke -a provision --how=container <...>
```

**Comment by lukaszachy** [comment]
Hm, I hit this several times too however `-a` is an option of `run` not the subcommands as `plans`. I'm not sure if it is a good idea to allow "global" options to be specified within subcommand options.
What we can easily fix is to use `--all` instead of `-a`. Same option but better readability.

**Comment by psss** [comment]
Yes, we should move the [Select Steps](https://tmt.readthedocs.io/en/stable/examples.html#select-steps) section from examples to the Guide and explicitly document the `--all` option there.

**Comment by psss** [comment]
The section should also mention the very frequent use case of listing tests which would be executed

    tmt run discover -v

Mentioned also in:

* #1067
